openwisp_monitoring/device/tests/test_models.py

```diff
@@ -578,6 +579,59 @@ def _create_env(self):
         )
         return dm, ping, load, process_count
 
+    def setUp(self):
+        from openwisp_monitoring.check.settings import CHECK_CLASSES
+
+        Check = load_model('check', 'Check')
+        self.device = self._create_device(organization=self._create_org())
+        self.ping_check = Check.objects.create(
+            name='Ping Check',
+            check_type=CHECK_CLASSES[0][0],
+            content_object=self.device,
+            params={},
+        )
+        self.device.monitoring.update_status('ok')
+
+    def test_check_signals(self):
+        with self.subTest('Test disabling a critical check'):
+            self.assertEqual(self.device.monitoring.status, 'ok')
+            with catch_signal(health_status_changed) as handler:
+                self.ping_check.is_active = False
+                self.ping_check.save()
+            self.assertEqual(handler.call_count, 1)
+            call_args = handler.call_args[0]
+            self.assertEqual(call_args[0], self.device.monitoring)
+            self.assertEqual(call_args[1], 'unknown')
+            self.device.refresh_from_db()
+            self.assertEqual(self.device.monitoring.status, 'unknown')
+            self.device.monitoring.update_status('ok')
```

Comment:
These sub-tests look good. I would just un-indent the part for resetting the DeviceMonitoring.status. So, the code gets executed regardless the subTest passes or fails. 

This applies to all the sub-tests.